hotel/web-application/src/main/java/com/andersenhotels/view/web_ui/rest/ReservationController.java

```diff
@@ -11,35 +11,39 @@
 @RequestMapping("/reservation")
 public class ReservationController {
 
-    private final ReservationRepository reservationRepository;
+    private final ReservationService reservationService;
 
-    public ReservationController(ReservationRepository reservationRepository) {
-        this.reservationRepository = reservationRepository;
+    public ReservationController(ReservationService reservationService) {
+        this.reservationService = reservationService;
     }
 
     @GetMapping
     public List<Reservation> getAllReservations() {
-        return reservationRepository.findAll();
+        return reservationService.getAllReservations();
     }
 
     @GetMapping("/{id}")
     public ResponseEntity<Reservation> getReservationById(@PathVariable long id) {
-        return reservationRepository.findById(id)
+        return reservationService.getReservationById(id)
                 .map(ResponseEntity::ok)
                 .orElseGet(() -> ResponseEntity.notFound().build());
     }
 
     @PostMapping
-    public Reservation createReservation(@RequestBody Reservation reservation) {
-        return reservationRepository.save(reservation);
+    public ResponseEntity<Reservation> createReservation(@RequestParam long apartmentId, @RequestParam long guestId) {
+        try {
+            Reservation createdReservation = reservationService.createReservation(apartmentId, guestId);
+            return ResponseEntity.ok(createdReservation);
+        } catch (IllegalArgumentException | IllegalStateException e) {
+            return ResponseEntity.badRequest().body(null);
+        }
```

Comment:
try catch better to use in service level
or use https://www.baeldung.com/exception-handling-for-rest-with-spring